### PR TITLE
enhancement(remap transform): add `parse_aws_vpc_flow_log` function

### LINKED ISSUE #5366: New `parse_aws_vpc_flow_log`
Same as #5365 but for [AWS VPC Flow logs](https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs.html)

### PR DESCRIPTION
**Pull request by fanatid**
Closes #5366

### CODE REVIEW

File: docs/reference/remap/parse_aws_vpc_flow.cue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+					"srcport":      null
+					"dstport":      null
+					"protocol":     null
+					"packets":      null
+					"bytes":        null
+					"start":        1431280876
+					"end":          1431280934
+					"action":       null
+					"log_status":   "NODATA"
+				}
```
Comment (fanatid) [inline docs/reference/remap/parse_aws_vpc_flow.cue:49]: I'm not sure here, do we need to add the key to the map if the value is `null`? @JeanMertz

Comment (JeanMertz) [reply docs/reference/remap/parse_aws_vpc_flow.cue:49]: I don't have a good answer.

My initial response would be "no", but if AWS does it, then we probably should as well, as that's what people expect from their output, maybe? Maybe our resident AWS expert @jszwedko has some thoughts on this.

Comment (jszwedko) [reply docs/reference/remap/parse_aws_vpc_flow.cue:49]: I think people will expect a consistent set of fields to be defined when parsing these logs so setting to `null` makes sense.

**Review by FungusHumungus — CHANGES_REQUESTED** [review]
Yeah, looks nice. 

Just one thing, it needs a behaviour test (https://github.com/timberio/vector/blob/master/tests/behavior/transforms/remap.toml).

These files are being moved into a separate crate in #5522, so we need to make sure this doesn't get lost in the move. The behaviour test will help with that!

**Comment by fanatid** [comment]
@jszwedko should we rename this function to `parse_aws_vpc_flow_log` (as `parse_aws_alb_log` from #5489)?

**Review by FungusHumungus — APPROVED** [review]
Nice.

**Review by jszwedko — COMMENTED** [review]
Maybe we should call this `parse_aws_vpc_flow_log` for parity with `parse_aws_alb_log`? Otherwise this looks good to me!

**Pull request merged**